<launch>
    <param name="use_sim_time" value="true"/>


<!-- load robot_description of k_crane -->
	<include file="$(find k_crane_description)/launch/k_crane_description_display_with_imu_lidar_using_rotate_joint.launch"/>
      <node type="laser_scan_assembler" pkg="laser_assembler" name="laser_assembler" output="screen" required="true">
            <!-- <remap from="scan" to="/scan_filtered"/> -->
            <remap from="scan" to="/ld_lrs3611/scan"/>
            <param name="max_scans" type="int" value="8000" />
            <param name="fixed_frame" type="string" value="world" />
      </node>

      <node type="LaserAssembler.py" pkg="laser_assembler"  name="LaserAssembler" output="screen"  required="true"/>  



      <node type="odom_crane_structural_info_republisher_in_world_frame_using_quat.py" pkg="trajectory_analysis"  name="combine_tf_frames_and_publish_odom_of_crane_structural"  output="screen" required="true"/> 
      <node type="odom_vins_republisher_in_world_frame_using_quat.py" pkg="trajectory_analysis"  name="odom_vins_republisher_in_world_frame_using_quat"  output="screen" required="true"/> 


      <node type="tf2_broadcaster_kobelco_2022.py" pkg="laser_to_pcl"  name="tf2_broadcaster_kobelco_2022"  output="screen" required="true"/>  
      <!-- <node type="tf2_broadcaster_kobelco_2022_real_sense.py" pkg="laser_to_pcl"  name="tf2_broadcaster_kobelco_2022_real_sense"  output="screen" required="true"/>   -->
      <!-- <include file="$(find vins_estimator)/launch/crane_realsense_fisheye_t265.launch" /> -->



      <node name="rviz" pkg="rviz" type="rviz" args="-d $(find realsense2_camera)/rviz/t265_model_crane.rviz" required="true" />

      <node name="tf_to_path_realsense" pkg="tf_to_trajectory" type="tf_to_path.py">
            <param name="parent_frame_id" value="camera_odom_frame"/>
            <param name="child_frame_id" value="camera_pose_frame"/>
            <param name="out_topic_name" value="/traj_by_realsense"/>
            <param name="append_rate" value="5"/>
      </node>


      <node name="tf_to_path_real_sense" pkg="tf_to_trajectory" type="odom_to_path.py">
            <param name="frame_id" value="world"/>
            <param name="topic_name" value="/camera/odom/sample"/>
            <param name="out_topic_name" value="/camera/odom/sample_path"/>
            <param name="append_rate" value="5"/>
      </node>



      <node name="tf_to_path_crane_info" pkg="tf_to_trajectory" type="tf_to_path.py">
            <param name="parent_frame_id" value="world"/>
            <param name="child_frame_id" value="crane_info"/>
            <param name="out_topic_name" value="/traj_by_crane_structure"/>
            <param name="append_rate" value="5"/>
      </node>



      <node name="tf_to_path_odometry_vins" pkg="tf_to_trajectory" type="odom_to_path.py">
            <param name="frame_id" value="world"/>
            <!-- <param name="topic_name" value="/vins_estimator/odometry"/> -->
            <param name="topic_name" value="/vins"/>
            <param name="out_topic_name" value="/vins_estimator/odometry_path"/>
            <param name="append_rate" value="5"/>
      </node>




</launch>




